src/preact/component/intersection-observer.js

```diff
@@ -13,17 +10,19 @@ import {useCallback, useRef} from '#preact';
  */
 export function useIntersectionObserver(callback) {
   const nodeRef = useRef(null);
```

Comment:
nit: do we still need `nodeRef` and `prevNode` now that cleanup is handled via `unobserveRef`?